examples/howto/data_container.myst.md

```diff
@@ -108,82 +168,177 @@ Coordinates are automatically stored into the {class}`arviz.InferenceData` objec
 idata.posterior.coords
 ```
 
+Coordinates are also used by `arviz` when making plots. Here we pass `legend=True` to `az.plot_trace` to automatically label the three cities in our trace plot.
+
 ```{code-cell} ipython3
-az.plot_trace(idata, var_names=["europe_mean_temp", "city_temperature"]);
+axes = az.plot_trace(idata, var_names=["europe_mean_temp", "expected_city_temp"], legend=True);
 ```
 
-We can get the data container variable from the model using:
+When we use `pm.ConstantData`, the data are internally represented as a pytensor `TensorConstant`.
 
 ```{code-cell} ipython3
-model["data"].get_value()
+type(data)
 ```
 
-Note that we used a theano method {meth}`theano.compile.sharedvalue.SharedVariable.get_value` of class {class}`theano.compile.sharedvalue.SharedVariable` to get the value of the variable. This is because our variable is actually a `SharedVariable`.
+If you need to, you can view the data using the `data` method. All `PyMC` variables, including `Data` containers, can be accessed by indexing the model object with a variable name. Since this line:
+
+```python
+    data = pm.ConstantData("observed_temp", df_data, dims=("date", "city"))
+```
+
+Gave the name "observed_temp" to the data, we can access it as follows: 
 
 ```{code-cell} ipython3
-type(data)
+model["observed_temp"].data[:15]
 ```
 
-The methods and functions related to the Data container class are:
+## MutableData
 
-- `data_container.get_value` (method inherited from the theano SharedVariable): gets the value associated with the `data_container`.
-- `data_container.set_value` (method inherited from the theano SharedVariable): sets the value associated with the `data_container`.
-- {func}`pymc.set_data`: PyMC3 function that sets the value associated with each Data container variable indicated in the dictionary `new_data` with it corresponding new value.
+In many cases, you will want the ability to switch out data between model runs. This arises when you want to fit a model to multiple datasets, or if you are interested in out-of-sample prediction. For these cases, `pm.MutableData` is the correct tool.
 
 +++
 
-## Using Data container variables to fit the same model to several datasets
+### Using MutableData container variables to fit the same model to several datasets
 
-This and the next sections are an adaptation of the notebook ["Advanced usage of Theano in PyMC3"](../Advanced_usage_of_Theano_in_PyMC3.html#using-shared-variables) using `pm.Data`.
-
-We can use `Data` container variables in PyMC3 to fit the same model to several datasets without the need to recreate the model each time (which can be time consuming if the number of datasets is large):
+We can use `MutableData` container variables in PyMC to fit the same model to several datasets without the need to recreate the model each time (which can be time consuming if the number of datasets is large):
```

Comment:
Unfortunately we recompile the d/logp function everything `pm.sample` is called so this is no longer true. It used to be true because we used to cache them

Reply:
I still think the sentence is true, you're just taking a very strict view of the meaning of "recreate" to mean "recompile". It would still be time consuming to copy/paste a large model over and over to re-use it on multiple datasets.